azurelinuxagent/common/protocol/wire.py

```diff
@@ -743,10 +754,7 @@ def _fetch_response(self, uri, headers=None, use_proxy=None, max_retry=None, ret
             msg = "Fetch failed: {0}".format(error)
             logger.warn(msg)
             report_event(op=WALAEventOperation.HttpGet, is_success=False, message=msg, log_event=False)
-
```

Comment:
This method use to return None for errors, or raise an exception for InvalidContainerError and ResourceGoneError. Now it always reports errors with exceptions